bin/varnishd/cache/cache_fetch.c

```diff
@@ -450,6 +450,10 @@ vbf_stp_startfetch(struct worker *wrk, struct busyobj *bo)
 	if (http_IsStatus(bo->beresp, 304) && vbf_304_logic(bo) < 0)
 		return (F_STP_ERROR);
 
+	if (bo->htc->doclose == SC_NULL &&
+	    http_GetHdrField(bo->bereq, H_Connection, "close", NULL))
+		bo->htc->doclose = SC_REQ_CLOSE;
+
 	VCL_backend_response_method(bo->vcl, wrk, NULL, bo, NULL);
 
```

Comment:
... and shouldn't the check for the beresp Connection header happen here?

Reply:
No preference, I'll go with the consensus. At the very least the case where the backend sends a `Connection: close` and we overwrite it is covered.

Reply:
On second thought I find it cleaner to have the distinction between `REQ_CLOSE` and `RESP_CLOSE`, and to avoid touching `beresp` headers `vbe_dir_finish()`.

Done.